### PR TITLE
chore(@clayui/css): Use Reuse Spec Snippets for licensing code snippets from Normalize.css, Suitcss, and HTML5 Boilerplate

### LINKED ISSUE #3437: Update Licensing for Normalize and HTML5 Boilerplate used in _reboot.scss and _print.scss
Bootstrap 4 uses:
- Normalize in https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/components/_reboot.scss
- HTML5 Boilerplate in https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/components/_print.scss

We should update our license text to attribute using the SPDX naming scheme.

### PR DESCRIPTION
**Pull request by pat270**
- moves license text to its own file so we don't overwrite commit history every time we release a new version
- removes Clay version from individual svg source files for same reason

fixes #3437

@bryceosterhaus @wincent

### CODE REVIEW

File: packages/clay-css/src/scss/components/_print.scss [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
-
-// ==========================================================================
-// Print styles.
-// Inlined to avoid the additional HTTP request:
-// https://www.phpied.com/delay-loading-your-print-css/
-// ==========================================================================
+/* REUSE-Snippet-Begin
+* SPDX-License-Identifier: MIT
+* SPDX-FileCopyrightText: © 2018 HTML5 Boilerplate <https://github.com/h5bp/main.css>
+*/
```
Comment (wincent) [inline packages/clay-css/src/scss/components/_print.scss:4]: This looks whitespace damaged; would expect:

```
/* REUSE-Snippet-Begin
 * SPDX-License-Identifier: MIT
 * etc
```

File: packages/clay-css/src/scss/components/_reboot.scss [hunk 2]
```diff
@@ -1,9 +1,7 @@
-// Reboot
-//
-// Normalization of HTML elements, manually forked from Normalize.css to remove
-// styles targeting irrelevant browsers while applying new styles.
-//
-// Normalize is licensed MIT. https://github.com/necolas/normalize.css
+/* REUSE-Snippet-Begin
+* SPDX-License-Identifier: MIT
+* SPDX-FileCopyrightText: © 2016 Nicolas Gallagher and Jonathan Neal <https://github.com/necolas/normalize.css>
+*/
```
Comment (wincent) [inline packages/clay-css/src/scss/components/_reboot.scss:4]: Same here with leading whitespace.

File: packages/clay-css/src/scss/components/_utilities-functional-important.scss [hunk 3]
```diff
@@ -188,8 +188,10 @@
 	}
 }
 
-// Embed
-// Credit: Nicolas Gallagher and SUIT CSS.
+/* REUSE-Snippet-Begin
+* SPDX-License-Identifier: MIT
+* SPDX-Copyright: © 2012 Nicolas Gallagher <https://github.com/suitcss/components-flex-embed>
```
Comment (wincent) [inline packages/clay-css/src/scss/components/_utilities-functional-important.scss:193]: And here (leading whitespace).

File: packages/clay-css/gulpfile.js [hunk 4]
```diff
@@ -231,11 +227,22 @@ gulp.task('version', function() {
 		}
 	});
 
-	fs.readdirSync(svgsDir).forEach(function(file) {
-		var filePath = path.join(svgsDir, file);
+	var libIconsSvg = path.join('.', 'lib', 'images', 'icons', 'icons.svg');
+
+	var regex = /<\?xml version=\"1.0\" encoding=\"UTF-8\"\?([\s\S]+-->|>)/g;
```
Comment (wincent) [inline packages/clay-css/gulpfile.js:232]: Pro-Tip™: you can use [the `s` flag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/dotAll) to make `.` match everything, including newlines, which means you can get rid of the `[\s\S]` construct:


```js
var regex = /<\?xml version=\"1.0\" encoding=\"UTF-8\"\?(.+-->|>)/gs;
```

Also, you have some excess escaping in there:

```js
var regex = /<\?xml version="1.0" encoding="UTF-8"\?(.+-->|>)/gs;
```

File: packages/clay-css/src/scss/_license-text.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+*/
+
+/**
+* Bootstrap v4.4.1
+*
+* SPDX-FileCopyrightText: © 2019 Twitter, Inc. <https://twitter.com>
+* SPDX-FileCopyrightText: © 2019 The Bootstrap Authors <https://getbootstrap.com/>
+*
+* SPDX-License-Identifier: LicenseRef-MIT-Bootstrap
+*/
```
Comment (wincent) [inline packages/clay-css/src/scss/_license-text.scss:17]: This all looks whitespace damaged to me:

```scss
/**
 * Clay 3.13.0
 *
```

etc.

File: packages/clay-css/gulpfile.js [hunk 6]
```diff
@@ -199,23 +199,19 @@ gulp.task('version', function() {
...
 	var clayFiles = [
-		path.join(scssDir, 'atlas-variables.scss'),
-		path.join(scssDir, 'atlas.scss'),
-		path.join(scssDir, 'base-variables.scss'),
-		path.join(scssDir, 'base.scss'),
+		path.join(scssDir, '_license-text.scss'),
 	];
 
-	function changeVersion(filePath) {
+	function changeVersion(filePath, regex = /\*\sClay\s(.*)\n/g, value = `* Clay ${VERSION}\n`) {
```
Comment (wincent) [inline packages/clay-css/gulpfile.js:208]: Slightly more robust regex would be:

```
/\*\s+Clay\s(.+)\n/g
```

File: packages/clay-css/src/scss/components/_print.scss (hunk 1 above)
Comment (pat270) [reply packages/clay-css/src/scss/components/_print.scss:4]: sent a SF pull at https://github.com/liferay/clay/pull/3485

File: packages/clay-css/gulpfile.js (hunk 4 above)
Comment (pat270) [reply packages/clay-css/gulpfile.js:232]: Thanks for the tips! Regex is a mystery to me. I think I understand `\s+` below, but I don't understand the difference between `.*` and `.+`. `.*` matches 0 or more of the preceding token and `.+` matches 1 or more of the preceding token. Where would `.*` match something I wouldn't want in this example vs `.+`?

Comment (wincent) [reply packages/clay-css/gulpfile.js:232]: Original: `/\*\sClay\s(.*)\n/g`

Would match: `* Clay 4.0.0\n` but also `* Clay \n`

Proposed: `/\*\s+Clay\s(.+)\n/g`

Would only match: `* Clay 4.0.0\n`

In other words, the second one is more robust because it is less likely to run into a false positive if it sees a line like the second one here:

```
 * see also the documentation for
 * Clay 
```

Of course, we're talking about an unlikely edge case, but the art of this is being pragmatic: make stuff as robust as possible *at a reasonable cost*, but no more.

There's [that saying](https://en.wikipedia.org/wiki/Robustness_principle) about "be liberal in what you accept and conservative in what you provide", which, in the world of regexps, translates to something like this:

- Be "loose" so that you don't falsely reject inputs that could be valid. An example of that would be changing the `\s` to `\s*` or `\s+` in `/\*\s\Clay/`; ie. don't reject something on the basis of the amount of whitespace in an insignificant position.
- But be "strict" so that you don't accept obviously incorrect input. An example of that would be making something that matches version numbers instead of just `(.+)`. That could be something as simlpe as `/\d\.d\.\d/` or as complicated as `/(\d{1,2}\.\d{1,2}(?:\.\d{1,2})?(?:-.+)?)/`, or more...

Basically, it's all very subjective, and I think over time you develop a gut instinct for when it's time to take a step in the direction of strictness or of looseness, because you spider sense starts tingling about false positives or false negatives. There's also the trade-off on the axis of readability too (like those version number examples above); here again it's a largely subjective judgement about what is "optimal" — amusingly "optimal" doesn't mean "perfect", but rather "good enough". 😂
